site/content/docs/02-routing.md

```diff
@@ -0,0 +1,127 @@
+
+
+### Pages
+
+Pages are Svelte components written in `.svelte` files. When a user first visits the application, they will be served a server-rendered version of the route in question, plus some JavaScript that 'hydrates' the page and initialises a client-side router. From that point forward, navigating to other pages is handled entirely on the client for a fast, app-like feel.
+
+The filename determines the route. For example, `src/routes/index.svelte` is the root of your site:
+
+```html
+<!-- src/routes/index.svelte -->
+<svelte:head>
+	<title>Welcome</title>
+</svelte:head>
+
+<h1>Hello and welcome to my site!</h1>
+```
+
+A file called either `src/routes/about.svelte` or `src/routes/about/index.svelte` would correspond to the `/about` route:
+
+```html
+<!-- src/routes/about.svelte -->
+<svelte:head>
+	<title>About</title>
+</svelte:head>
+
+<h1>About this site</h1>
+<p>TODO...</p>
+```
+
+Dynamic parameters are encoded using `[brackets]`. For example, here's how you could create a page that renders a blog post:
+
+```html
+<!-- src/routes/blog/[slug].svelte -->
+<script context="module">
+	// the (optional) preload function takes a
+	// `{ path, params, query }` object and turns it into
+	// the data we need to render the page
+	export async function preload(page, session) {
+		// the `slug` parameter is available because this file
+		// is called [slug].svelte
+		const { slug } = page.params;
+
+		// `this.fetch` is a wrapper around `fetch` that allows
+		// you to make credentialled requests on both
+		// server and client
+		const res = await this.fetch(`blog/${slug}.json`);
+		const article = await res.json();
+
+		return { article };
+	}
+</script>
+
+<script>
+	export let article;
+</script>
+
+<svelte:head>
+	<title>{article.title}</title>
+</svelte:head>
+
+<h1>{article.title}</h1>
+
+<div class='content'>
+	{@html article.html}
+</div>
+```
+
+> See the section on [preloading](docs#preloading) for more info about `preload` and `this.fetch`
+
+
+### Server routes
+
+Server routes are modules written in `.js` files that export functions corresponding to HTTP methods. Each function receives HTTP `request` and `response` objects as arguments, plus a `next` function. This is useful for creating a JSON API. For example, here's how you could create an endpoint that served the blog page above:
+
+```js
+// routes/blog/[slug].json.js
+import db from './_database.js'; // the underscore tells Sapper this isn't a route
+
+export async function get(req, res, next) {
+	// the `slug` parameter is available because this file
+	// is called [slug].json.js
+	const { slug } = req.params;
+
+	const article = await db.get(slug);
+
+	if (article !== null) {
+		res.setHeader('Content-Type', 'application/json');
+		res.end(JSON.stringify(article));
+	} else {
+		next();
+	}
+}
+```
+
+> `delete` is a reserved word in JavaScript. To handle DELETE requests, export a function called `del` instead.
```

Comment:
The docs for this is absolutely the wrong place to be posting this, but: In line with Svelte 'export-as' thing, might we want to do the same thing here rather than treating a 'del' export specially?

Reply:
I think it's probably easier just to export `del` — either way, you need to know about `delete` being reserved, so export-as only adds extra code:

```js
export function del() {...}

// vs
function del() {...}
export { del as delete }
```